ably/params.go

```diff
@@ -0,0 +1,137 @@
+package ably
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type ProtocolType string
+
+const (
+	ProtocolJSON    = "json"
+	ProtocolMsgPack = "msgpack"
+)
+
+type Params struct {
```

Comment:
Params is quite generic, should this not perhaps be named in such a way that a developer knows what the `Params` are for?